src/dmd/expression.d

```diff
@@ -6612,7 +6614,8 @@ extern (C++) final class CondExp : BinExp
 
                     if (v.needsScopeDtor())
                     {
-                        if (!vcond)
+                        auto cve = ce.econd.isVarExp();
+                        if (!vcond && !cve)
```

Comment:
Doesn't seem relevant here either.

Reply:
This was needed because it caused errors when copying `__ctfe` to a temporary variable. I could have simply excluded that case, but I chose the more general approach, where I excluded every simple variable, as they don't need to be saved as temporaries.

Another approach would have been to use `extractSideEffect` instead of `copyToTemp` in the code below. `extractSideEffect` would leave the cond unchanged if it's trivial and has no side effects. Would this work better?